BREAKING CHANGE: xCertificate renamed to CertificateDsc: Renamed all DSCResources, Examples, Modules and Tests where applicable

timhaintz opened the pull request:
<!--
Thanks for submitting a Pull Request (PR) to this project. Your contribution to this project is greatly appreciated!

Please prefix the PR title with the resource name, i.e. 'xCertReq: My short description'
If this is a breaking change, then also prefix the PR title with 'BREAKING CHANGE:', i.e. 'BREAKING CHANGE: xCertReq: My short description'

To aid community reviewers in reviewing and merging your PR, please take the time to run through the below checklist.
Change to [x] for each task in the task list that applies to this PR.
-->
**Pull Request (PR) description**
<!-- Replace this with a description of your pull request -->

**This Pull Request (PR) fixes the following issues:**
<!-- Replace this with the list of issues or n/a. Use format: Fixes #123 -->

**Task list:**
- [ x] Change details added to Unreleased section of CHANGELOG.md?
- [x ] Added/updated documentation, comment-based help and descriptions in .schema.mof files where appropriate?
- [ x] Examples appropriately updated?
- [ x] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [x ] [Unit and (optional) Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

timhaintz commented:
@PlagueHO CertificateDsc\Modules\CertificateDsc\DSCResources\MSFT_CertReq\MSFT_CertReq.psm1 VSCode is telling me that the variable $rspPath isn't used on line 420. Did you want to leave it in there?
Thanks. :D

PlagueHO commented:
@timhaintz - yeah, go on - get rid of it :grin: make a note in the CHANGELOG that you got rid of unnecessary code though. I was originally planned to use the Response file but then never did.

I'll should have some time for a review tonight! I might actually pick up those stalled PR's and make them myself as they just needed minor tweaks to get through.

timhaintz commented:
Thanks @PlagueHO all done. Let me know what I have missed and I will fix it up. 

If you have any other things you would like me to look at, happy to help out. Flick it my way and I will try and work it out. Thanks heaps. 😃

PlagueHO commented:
Sorry about the delay @timhaintz - I've been off sick since the summit and a bit over saturated at work. I'll try and catch up this weekend.

timhaintz commented:
No worries @PlagueHO . Hope you're feeling better. I'm working a bit over the weekend so no hurry from here. 😄

PlagueHO commented:
Awesome job @timhaintz - faultless! Could you just make a mention of the removal of that RSP assignment line in the readme.md and I think we're then ready to merge.

---

Reviewed 108 of 147 files at r1, 1 of 3 files at r2.
Review status: 59 of 60 files reviewed at latest revision, all discussions resolved.

---

*[CHANGELOG.md, line 13 at r2](https://beta.reviewable.io/reviews/powershell/xcertificate/124#-L8BdflL1kiwMAJwlHkM:-L8BdflL1kiwMAJwlHkN:b-fppp7e) ([raw file](https://github.com/powershell/xcertificate/blob/3dcba98b6d67ca006d7e3fc0bcc7fa73ebb702ec/CHANGELOG.md#L13)):*
> ```Markdown
>   - Removed unnecessary code from:
>     - CertificateDsc\Modules\CertificateDsc\DSCResources\MSFT_CertReq\MSFT_CertReq.psm1
> ```

Could you mention removal of that redundant line in xCertReq (the one assigning the filename of the .RSP)?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xcertificate/124#-:-L8BdpKP4lgJ0A9-fw2X:bqhknl1)*
<!-- Sent from Reviewable.io -->

timhaintz commented:
So close.... Nearly got it without any updates the first iteration. 😃 . Updated CHANGELOG.md to make it clearer what I deleted. Also updated README.md. There wasn't detailed into in the other README.md files. Put the month I deleted it so someone could come back to this commit and check where it was.

Hope you're feeling better @PlagueHO . Thanks!

PlagueHO commented:
Reviewed 1 of 2 files at r3.
Review status: 59 of 60 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md, line 6 at r4](https://reviewable.io:443/reviews/powershell/xcertificate/124#-L8JrWNueNEgFdhzCuxf:-L8JrWNw4rwhjo_KYlPK:b-5iehj1) ([raw file](https://github.com/powershell/xcertificate/blob/4f09d965fb21e860f63941bc52dd83e018e892e8/Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md#L6)):*
> ```Markdown
> authority.
> 
> - March 2018
> ```

I don't think this is needed here and might just confuse people. So might be worth removing this bit. Otherwise people might expect to see all changes in appear in this file. :cry: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/124)*
<!-- Sent from Reviewable.io -->

timhaintz commented:
Review status: 59 of 60 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md, line 6 at r4](https://reviewable.io:443/reviews/powershell/xcertificate/124#-L8JrWNueNEgFdhzCuxf:-L8WnM5Ep4FWrRA0KA0p:bpuln5f) ([raw file](https://github.com/powershell/xcertificate/blob/4f09d965fb21e860f63941bc52dd83e018e892e8/Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md#L6)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I don't think this is needed here and might just confuse people. So might be worth removing this bit. Otherwise people might expect to see all changes in appear in this file. :cry: 
</blockquote></details>

Sorry, got confused with what you were looking for. Do you want me to remove the March part or all of it? I thought in one of your earlier comments you were asking me to update README.md, I might have been off track though. Let me know what you would like and I will update it. Sorry for the delay too. Been a bit on a work. :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/124)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 59 of 60 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md, line 6 at r4](https://beta.reviewable.io/reviews/powershell/xcertificate/124#-L8JrWNueNEgFdhzCuxf:-L8nxshf7eUz9z5qd_T3:b-5kj4eg) ([raw file](https://github.com/powershell/xcertificate/blob/4f09d965fb21e860f63941bc52dd83e018e892e8/Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md#L6)):*
<details><summary><i>Previously, timhaintz wrote…</i></summary><blockquote>

Sorry, got confused with what you were looking for. Do you want me to remove the March part or all of it? I thought in one of your earlier comments you were asking me to update README.md, I might have been off track though. Let me know what you would like and I will update it. Sorry for the delay too. Been a bit on a work. :)
</blockquote></details>

No worries at all - I'm just so glad of the help - and as you can see I've been a bit sidetracked! Anyway, I think we should remove this and just leave the stuff in the CHAGNGELOG.MD. Again, thanks for your help Tim! 

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xcertificate/124)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Reviewed 1 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xcertificate/124)*
<!-- Sent from Reviewable.io -->

timhaintz commented:
Review status: 59 of 60 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md, line 6 at r4](https://beta.reviewable.io/reviews/powershell/xcertificate/124#-L8JrWNueNEgFdhzCuxf:-L8otlI-11QKMhItjAtg:b-5k10dl) ([raw file](https://github.com/powershell/xcertificate/blob/4f09d965fb21e860f63941bc52dd83e018e892e8/Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md#L6)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

No worries at all - I'm just so glad of the help - and as you can see I've been a bit sidetracked! Anyway, I think we should remove this and just leave the stuff in the CHAGNGELOG.MD. Again, thanks for your help Tim! 
</blockquote></details>

README.md updated/removed content. Thanks for all of your help and patience. :+1: Will have a look around some more of your repo's and hopefully help some more. Happy to help. Thanks. :smile: 

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xcertificate/124)*
<!-- Sent from Reviewable.io -->

timhaintz commented:
Review status: 59 of 60 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md, line 6 at r4](https://beta.reviewable.io/reviews/powershell/xcertificate/124#-L8JrWNueNEgFdhzCuxf:-L8oueCgC-BS1WAuTkLT:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/4f09d965fb21e860f63941bc52dd83e018e892e8/Modules/CertificateDsc/DSCResources/MSFT_CertReq/README.md#L6)):*
<details><summary><i>Previously, timhaintz wrote…</i></summary><blockquote>

README.md updated/removed content. Thanks for all of your help and patience. :+1: Will have a look around some more of your repo's and hopefully help some more. Happy to help. Thanks. :smile: 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xcertificate/124)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Awesome! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

@KwirkyKat - can you please do the rename of this repo over to CertificateDsc for us when you're able? Thank you!

---

Reviewed 1 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/xcertificate/124#-:-L8sul8H5oGJyrMBiwXT:b-y6z43w)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@PlagueHO No idea why, but GH won't let me do the rename. I keep getting "Something went wrong". Tried different browsers but it didn't matter. Probably going to have to open a support ticket with GH
![image](https://user-images.githubusercontent.com/6372220/38340521-caf7dbf8-3828-11e8-8228-4cfcd8e77bdf.png)

kwirkykat commented:
@PlagueHO Seems to be happening on all repos. I think someone changed the PowerShell org default permissions settings. Will update when I know anything else

PlagueHO commented:
Thanks @kwirkykat !

kwirkykat commented:
@PlagueHO Worked now. GitHub must have fixed something on their end.

johlju commented:
@PlagueHO Can you update the DscResources repo as well, switching the submodules folder?

PlagueHO commented:
Done - in my existing PR adding the submodules
